hydra-node/test/Hydra/BehaviorSpec.hs

```diff
@@ -479,9 +480,9 @@ spec = parallel $ do
             -- We expect one Commit AND the CollectCom to be rolled back and
             -- forward again
             rollbackAndForward chain 2
-            waitUntil [n1] RolledBack
-            waitUntil [n1] $ Committed testHeadId alice (utxoRef 1)
-            waitUntil [n1] $ HeadIsOpen{headId = testHeadId, utxo = utxoRefs [1]}
+            -- We expect the node to still work and let us post L2 transactions
```

Comment:
I would rather say that we don't really care, in this test, if `HeadIsOpen` is emitted again. What we want to assert is that, once we've seen a head open, it's open and we can use it, no matter if there has been rollbacks.

But that's a good point because, looking at the HeadLogic, I don't like the smell of it. Eventhough the behavior would probably be fine, I think we can see `InvalidEvent` errors when the collectCom transaction is observed again but the HeadLogic consider the head as open already.

Reply:
We have the same taste of code then :)

Having `InvalidEvent` error outcomes now because we see a `OnCommit` or `OnCollectCom` when the head is already `Open`, is a smell indeed. I mentioned once that in this case we would need to change the head logic in a way that "the chain is always right" and we could identify already that "a rollback must have happened, we bet it all is fine though and stay tuned" is rather the outcome when seeing this event?